alerts.tf

```diff
@@ -0,0 +1,56 @@
+module "div-bad-requests-alert" {
+  source = "git@github.com:hmcts/cnp-module-metric-alert"
+  location = "${var.location}"
+
+  app_insights_name = "div-${var.env}"
+
+  alert_name = "div-bad-requests"
+  alert_desc = "Found HTTP requests with 400 error response code (bad request) in div-${var.env}."
+  app_insights_query = "requests | where resultCode == \"400\""
+  custom_email_subject = "Alert: bad requests in div-${var.env}"
+  frequency_in_minutes = 5
+  time_window_in_minutes = 5
+  severity_level = "2"
+  action_group_name = "Divroce Support"
+  trigger_threshold_operator = "GreaterThan"
+  trigger_threshold = 1
+  resourcegroup_name = "${azurerm_resource_group.rg.name}"
+}
+
+module "div-server-errors-alert" {
+  source = "git@github.com:hmcts/cnp-module-metric-alert"
+  location = "${var.location}"
+
+  app_insights_name = "div-${var.env}"
+
+  alert_name = "div-server-errors"
+  alert_desc = "Found HTTP requests with 400 error response code (bad request) in div-${var.env}."
+  app_insights_query = "requests | where resultCode startswith \"5\"  | where name !contains \"/health\""
+  custom_email_subject = "Alert: server errors in div-${var.env}"
+  frequency_in_minutes = 5
+  time_window_in_minutes = 5
+  severity_level = "2"
+  action_group_name = "Divroce Support"
+  trigger_threshold_operator = "GreaterThan"
+  trigger_threshold = 5
+  resourcegroup_name = "${azurerm_resource_group.rg.name}"
+}
+
+module "div-fe-performance-alert" {
+  source = "git@github.com:hmcts/cnp-module-metric-alert"
+  location = "${var.location}"
+
+  app_insights_name = "div-${var.env}"
+
+  alert_name = "div-fe-performance-alert"
+  alert_desc = "Web pages took longer than 10 seconds to load in div-${var.env}."
```

Comment:
Not really sure if this alert will be really useful, is frequent having request slower than 10 secs, (12 in last 24h). Azure has alert when the performance has degraded, We received them on aat, probably devops received them from prod

Reply:
The minimum is 5 requests taking longer than 10 seconds within a 5 minute window. I had a look at the 12 you've mentioned and they're spread around so this wouldn't have triggered for those anyway.

I saw the Azure performance degraded emails you've mentioned, and they seem to refer to particular components.
This is a simple "Is website suddenly slower as a user for a long period of time E2E?" (think CCD slowness) 

We may have to tweak the frequency/threashold overtime if it becomes too noisy, but I think it's very valuable